tests/returntypes/test_jax_new.py

```diff
@@ -0,0 +1,1125 @@
+
+        def cost(a, cache):
+            with qml.tape.QuantumTape() as tape:
+                qml.RY(a[0], wires=0)
+                qml.RX(a[1], wires=0)
+                qml.RY(a[2], wires=0)
+                qml.expval(qml.PauliZ(0))
+                qml.expval(qml.PauliZ(1))
+
+            res = qml.interfaces.execute(
+                [tape], dev, cache=cache, interface="jax-python", **execute_kwargs
+            )
+            return res[0]
+
+        res = jax.jacobian(cost)(params, cache=None)
+
+        assert isinstance(res, tuple)
+        assert len(res) == 2
+
+        assert res[0].shape == (3,)
+        assert isinstance(res[0], jax.numpy.ndarray)
+
+        assert res[1].shape == (3,)
+        assert isinstance(res[1], jax.numpy.ndarray)
+
+    def test_multiple_expvals_single_par(self, execute_kwargs):
+        """Tests computing multiple expectation values in a tape with a single
+        trainable parameter."""
+        dev = qml.device("default.qubit", wires=2)
+        params = jnp.array([0.1])
+
+        def cost(a, cache):
+            with qml.tape.QuantumTape() as tape:
+                qml.RY(a[0], wires=0)
+                qml.expval(qml.PauliZ(0))
+                qml.expval(qml.PauliZ(1))
+
+            res = qml.interfaces.execute(
+                [tape], dev, cache=cache, interface="jax-python", **execute_kwargs
+            )
+            return res[0]
+
+        res = jax.jacobian(cost)(params, cache=None)
+
+        assert isinstance(res, tuple)
+
+        assert isinstance(res[0], jax.numpy.ndarray)
+        assert res[0].shape == (1,)
+
+        assert isinstance(res[1], jax.numpy.ndarray)
+        assert res[1].shape == (1,)
+
+    def test_multi_tape_fwd(self, execute_kwargs):
+        """Test the forward evaluation of a cost function that uses the output
+        of multiple tapes that be vector-valued."""
+        dev = qml.device("default.qubit", wires=2)
+        params = jax.numpy.array([0.3, 0.2])
+
+        def cost_fn(x):
+            with qml.tape.QuantumTape() as tape1:
+                qml.RX(x[0], wires=[0])
+                qml.expval(qml.PauliY(0))
+
+            with qml.tape.QuantumTape() as tape2:
+                qml.RX(x[1], wires=[0])
+                qml.RX(x[1], wires=[0])
+                qml.RX(-x[1], wires=[0])
+                qml.expval(qml.PauliY(0))
+                qml.expval(qml.PauliY(1))
+
+            result = qml.execute(
+                tapes=[tape1, tape2], device=dev, interface="jax", **execute_kwargs
+            )
+            return result[0] + result[1][0]
+
+        expected = -jnp.sin(params[0]) + -jnp.sin(params[1])
+        res = cost_fn(params)
+        assert jnp.allclose(expected, res)
+
+    # TODO: Add test after Autograd interface for return types is added
+    @pytest.mark.xfail(reason="Need autograd")
+    def test_multi_tape_jacobian(self, execute_kwargs):
+        """Test the jacobian computation with multiple tapes."""
```

Comment:
Since autograd integration is already merged, can this test and the following one be added back in?

Reply:
Yes thank you for noting that